rewrite-maven/src/main/java/org/openrewrite/maven/AddGradleEnterpriseMavenExtension.java

```diff
@@ -0,0 +1,187 @@
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.xml.*;
+import org.openrewrite.xml.tree.Xml;
+
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class AddGradleEnterpriseMavenExtension extends Recipe {
+    private static final String GRADLE_ENTERPRISE_MAVEN_EXTENSION_ARTIFACT_ID = "gradle-enterprise-maven-extension";
+    private static final String EXTENSIONS_XML_PATH = ".mvn/extensions.xml";
+    private static final String GRADLE_ENTERPRISE_XML_PATH = ".mvn/gradle-enterprise.xml";
+
+    @Language("xml")
+    private static final String EXTENSIONS_XML_FORMAT = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
+                                                        "<extensions>\n" +
+                                                        "</extensions>";
+
+    @Language("xml")
+    private static final String ENTERPRISE_TAG_FORMAT = "<extension>\n" +
+                                                       "  <groupId>com.gradle</groupId>\n" +
+                                                       "  <artifactId>gradle-enterprise-maven-extension</artifactId>\n" +
+                                                       "  <version>%s</version>\n" +
+                                                       "</extension>";
+
+    @Language("xml")
+    private static final String ENTERPRISE_TAG_FORMAT_WITHOUT_VERSION = "<extension>\n" +
+                                                       "  <groupId>com.gradle</groupId>\n" +
+                                                       "  <artifactId>gradle-enterprise-maven-extension</artifactId>\n" +
+                                                       "</extension>";
+
+    @Language("xml")
+    private static final String GRADLE_ENTERPRISE_XML_FORMAT = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\" ?>\n" +
+                                         "<gradleEnterprise\n" +
+                                         "    xmlns=\"https://www.gradle.com/gradle-enterprise-maven\" " +
+                                         "xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n" +
+                                         "    xsi:schemaLocation=\"https://www.gradle.com/gradle-enterprise-maven" +
+                                         " https://www.gradle.com/schema/gradle-enterprise-maven.xsd\">\n" +
+                                         "  <server>\n" +
+                                         "    <url>%s</url>\n" +
+                                         "    <allowUntrusted>%b</allowUntrusted>\n" +
+                                         "  </server>\n" +
+                                         "  <buildScan>\n" +
+                                         "    <backgroundBuildScanUpload>false</backgroundBuildScanUpload>\n" +
```

Comment:
Why is this hardcoded to `false`? Is this parameter required? If we're not making it configurable would rather not set a value for it unless required

Reply:
Yes, good question, I thought it was false by default, just asked the question [here](https://github.com/openrewrite/rewrite/issues/3005#issuecomment-1533593785)

Reply:
I saw @alextu is OOO until May 8, so maybe let us keep it as is (not sure if it's a required field and safe to remove it), and we can make another code update if we want to make it configurable.